src/util/redux.js

```diff
@@ -0,0 +1,235 @@
+	 * @param {Object} reducers - a tree of lucid reducers
+	 * @param {string[]} rootPath - array of strings representing the path to local state in global state
+	 * @returns {Object} action creator tree
+	 */
+	function createActionCreatorTree(reducers, rootPath) {
+		return (function recur(reducers, path = []) {
+			return _.reduce(reducers, (memo, node, key) => {
+				const currentPath = path.concat(key);
+				return _.assign(memo, {
+					[key]: _.isFunction(node) ?
+						createActionCreator(node, currentPath, rootPath) :
+						recur(node, currentPath)
+				});
+			}, {});
+		})(reducers, rootPath);
+	}
+
+	/**
+	 * getDispatchTree
+	 *
+	 * Walks the reducer tree and generates an action creator tree, then binds dispatch to each node
+	 * @param {Object} reducers - a tree of lucid reducers
+	 * @param {string[]} rootPath - array of strings representing the path to local state in global state
+	 * @param {function} dispatch - the redux store's `dispatch` function
+	 */
+	function getDispatchTree(reducers, rootPath, dispatch) {
+		const actionCreatorTree = createActionCreatorTree(reducers, rootPath);
+		dispatchTree = bindActionCreatorTree(actionCreatorTree, dispatch);
+		return dispatchTree;
+	}
+
+}
+
+/**
+ * createReduxReducerTree
+ *
+ * Walks the reducer tree and generates a tree of redux reducers, converting the
+ * signature from `(state, payload) => state` to `(state, action) => state`
+ * @param {Object} reducers - a tree of lucid reducers
+ * @param {string[]} path - array of strings representing the path to the reducer
+ * @return {Object} redux reducer tree
+ */
+function createReduxReducerTree(reducers, path = []) {
+	return _.reduce(reducers, (memo, node, key) => {
+		// filter out thunks from the reducer tree
+		if (node.isThunk) {
+			return memo;
+		}
+		const currentPath = path.concat(key);
+		return _.assign(memo, {
+			[key]: _.isFunction(node) ?
+				function reduxReducer(state, action) {
+					const { type, payload, meta = [] } = action;
+					if (_.isUndefined(state) || type !== currentPath.join('.')) {
+						return state;
+					}
+					return node(state, payload, ...meta);
+				} :
+				createReduxReducerTree(node, currentPath)
+		});
+	}, {});
+}
+
+/**
+ * createReducerFromReducerTree
+ *
+ * Returns a function that calls every reducer in the reducer tree with the reducer's local state and action
+ * @param {Object} reduxReducerTree - tree of redux reducers with signature `(state, action) => state`
+ * @param {Object} initialState - the initial state object that the reducer will return
+ * @return {function} the redux reducer
+ */
+function createReducerFromReducerTree(reduxReducerTree, initialState) {
+	return function reduxReducer(state, action) {
+		if (_.isUndefined(state)) {
+			return initialState;
+		}
+		return _.reduce(reduxReducerTree, (state, node, key) => {
+			return _.assign({}, state, _.isFunction(node) ? node(state, action) : {
```

Comment:
we're using object spread? My understanding is that it's not well-specced yet, so it's cool in jsx, but not yet js.